src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -110,7 +113,13 @@ internal ZipArchiveEntry(ZipArchive archive, String entryName)
             _originallyInArchive = false;
 
             _diskNumberStart = 0;
-
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
This library is currently compiled separately for Unix and Windows, so technically Unix vs. Windows could be determined at compile time instead of taking a dependency on `System.Runtime.InteropServices.RuntimeInformation`. But I'm guessing the use of `RuntimeInformation` is still preferred?